### PR TITLE
Adapt configuration for K8s API request timeouts

### PR DESCRIPTION
**Pull request by valiparsa**
### Description

In the previous version `0.18.0` default value of 1m is set in the Helmchart for `server-request-timeout` for both run- and tenant-controllers which are wrong. This release removes the default value of 1m in the Helmchart to be set by the consumer of the Helmchart.

### Submitter checklist

- [ ] Change has been tested (on a back-end cluster)
- [x] [changelog.yaml] entry with upgrade notes is prepared and appropriate for the audience affected by the change (users or developer, depending on the change)
- [x] Semantic version diffed against [last release][releases] and updated accordingly. In this project the version has to be maintained here:
    - [/charts/steward/Chart.yaml](https://github.com/SAP/stewardci-core/blob/master/charts/steward/Chart.yaml) (`version` and `appVersion`)

In case dependencies have been updated:
- N/A Links to external changelogs, since the last release of our component, added to the [changelog.yaml] entry (description).
- N/A Changelogs read thoroughly, potential impact described, upgrade notes prepared (if necessary)
- N/A Check if dependency updates affect our semantic version increment.

### Reviewer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There is at least 1 approval for the pull request and no outstanding requests for change
- [x] All voter checks have passed
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] The Pull Request title is understandable and reflects the changes well
- [x] The Pull Request description is understandable and well documented
- [x] [changelog.yaml] entry for this Pull Request has been added
    - [x] Changelog entry contains all required information
    - [x] 'Upgrade notes' are documented in changelog.yaml (if required)

[changelog.yaml]: https://github.com/SAP/stewardci-core/changelog.yaml
[releases]: https://github.com/SAP/stewardci-core/releases

### CODE REVIEW

File: charts/steward/values.yaml [hunk 1]
```diff
@@ -14,7 +14,6 @@ runController:
     heartbeatInterval: 1m
     heartbeatLogging: true
     heartbeatLogLevel: 3
-    serverRequestTimeout: 1m
```
Comment (rinckm) [inline charts/steward/values.yaml:17]: Keep it as documentation with some comment
```
# allowed values are durations. e.g. 1m 
serverRequestTimeout: ""
```

Comment (valiparsa) [reply charts/steward/values.yaml:17]: done in here https://github.com/SAP/stewardci-core/pull/307/commits/1e0917f87354a27e7ee8e1215eb4b5dacbda451c

**Review by rinckm — APPROVED** [review]

**Comment by anbrsap** [comment]
Could we change the option name from `serverRequestTimeout` to `k8sAPIRequestTimeout`? The former is rather unspecific, while the latter makes clear what is really meant.

File: charts/steward/values.yaml (hunk 1 above)
Comment (anbrsap) [reply charts/steward/values.yaml:17]: The documentation is in the README. So I don't think we should double-document parameter.

**Comment by valiparsa** [comment]
> k8sAPIRequestTimeout

That is good point. Thanks for your comment. I applied your suggestion in this commit:
https://github.com/SAP/stewardci-core/pull/307/commits/8301143aa14af904572d4a291cebe807eba235de

File: changelog.yaml [hunk 2]
```diff
@@ -52,6 +52,23 @@
...
+        - Rename command-line parameter `server-request-timeout` for the run controller to `k8s-api-request-timeout`
+        - Rename command-line parameter `server-request-timeout` for the tenant controller to `k8s-api-request-timeout`
+        - Rename Helm Chart values in the `values.yaml` file from `runController.serverRequestTimeout` to `runController.k8sAPIRequestTimeout`
+        - Rename Helm Chart values in the `values.yaml` file from `tenantController.serverRequestTimeout` to `tenantController.k8sAPIRequestTimeout`
+      upgradeNotes: |-
+        - In the previous version `0.18.0` default value of 1m is set in the Helmchart for `server-request-timeout` for both run- and tenant-controllers
+        which are wrong. This release removes the default value of 1m in the Helmchart to be set by the consumer of the Helmchart.
+        - Please rename `runController.serverRequestTimeout` to `runController.k8sAPIRequestTimeout` inside consumer's `values.yaml` file.
+        - Please rename `tenantController.serverRequestTimeout` to `tenantController.k8sAPIRequestTimeout` inside consumer's `values.yaml` file.
+      pullRequestNumber:
```
Comment (valiparsa) [inline changelog.yaml:70]:
```suggestion
      pullRequestNumber: 307
```

File: charts/steward/values.yaml (hunk 1 above)
Comment (anbrsap) [reply charts/steward/values.yaml:17]: Please remove the line.

File: charts/steward/values.yaml [hunk 3]
```diff
@@ -49,7 +50,8 @@ tenantController:
...
+    # allowed values are durations. e.g. 1m
```
Comment (anbrsap) [inline charts/steward/values.yaml:53]: Ditto

**Review by anbrsap — APPROVED** [review]

**Pull request merged**